web/front/app.js

```diff
@@ -154,6 +162,70 @@ export default {
       this.flamegraphdata = message;
       this.buildChart();
     },
+    getIdsFromChartNode(node) {
+      const { data } = node;
+      const { _inner } = data;
+      const { ids } = _inner;
+      return ids;
+    },
+    getMessageMapParentListFromId(id) {
+      let currentId = id;
+      const data = [currentId];
+      const findParentById = (id) => {
+        for (const parentId in msgmapchildren) {
+          for (const children of msgmapchildren[parentId]) {
+            if (children.id === id) {
+              return parentId;
+            }
+          }
+        }
+        return null;
+      }
+      while (true) {
```

Comment:
these are dangerous as they can become infinite - I usually use a counter with a large value to ensure a stop